tests/unit_tests/layout/test_layout_array.py

```diff
@@ -14,206 +14,248 @@
+        ntel_after = instance.get_number_of_telescopes()
+        assert ntel_before + 1 == ntel_after
 
-    layout.add_telescope(
-        telescope_name="LST-01",
-        crs_name="corsika",
-        xx=57.5 * u.m,
-        yy=57.5 * u.m,
-        tel_corsika_z=0 * u.m,
-    )
-    layout.add_telescope(
-        telescope_name="LST-02",
-        crs_name="corsika",
-        xx=-57.5 * u.m,
-        yy=57.5 * u.m,
-        tel_corsika_z=0 * u.m,
-    )
-    layout.add_telescope(
-        telescope_name="LST-02",
-        crs_name="corsika",
-        xx=57.5 * u.m,
-        yy=-57.5 * u.m,
-        tel_corsika_z=0 * u.m,
-    )
-    layout.add_telescope(
-        telescope_name="LST-04",
-        crs_name="corsika",
-        xx=-57.5 * u.m,
-        yy=-57.5 * u.m,
-        tel_corsika_z=0 * u.m,
-    )
+        instance.add_telescope("LST-00", "corsika", 100.0 * u.m, 50.0 * u.m, None, 50.0 * u.m)
+        assert instance._telescope_list[-1].get_altitude().value == pytest.approx(altitude)
 
-    layout.convert_coordinates()
-    layout.print_telescope_list()
-    layout.export_telescope_list(crs_name="corsika")
+    one_site(layout_array_north_instance, telescope_north_test_file, 2192.0)
+    one_site(layout_array_south_instance, telescope_south_test_file, 2181.0)
 
-    # Building a second layout from the file exported by the first one
-    layout_2 = LayoutArray(telescope_list_file=layout.telescope_list_file, name="test_layout_2")
 
-    assert 4 == layout_2.get_number_of_telescopes()
-    assert layout_2._array_center.get_altitude().value == pytest.approx(
-        layout._array_center.get_altitude().value, 1.0e-2
-    )
+def test_build_layout(
+    layout_array_north_four_LST_instance,
+    layout_array_south_four_LST_instance,
+    tmp_test_directory,
+    db_config,
+):
+    def one_site(layout, site_label):
+        layout.add_telescope(
+            telescope_name="LST-01",
+            crs_name="corsika",
+            xx=57.5 * u.m,
+            yy=57.5 * u.m,
+            tel_corsika_z=0 * u.m,
+        )
+        layout.add_telescope(
+            telescope_name="LST-02",
+            crs_name="corsika",
+            xx=-57.5 * u.m,
+            yy=57.5 * u.m,
+            tel_corsika_z=0 * u.m,
+        )
+        layout.add_telescope(
+            telescope_name="LST-02",
+            crs_name="corsika",
+            xx=57.5 * u.m,
+            yy=-57.5 * u.m,
+            tel_corsika_z=0 * u.m,
+        )
+        layout.add_telescope(
+            telescope_name="LST-04",
+            crs_name="corsika",
+            xx=-57.5 * u.m,
+            yy=-57.5 * u.m,
+            tel_corsika_z=0 * u.m,
+        )
+
+        layout.convert_coordinates()
+        layout.print_telescope_list()
+        layout.export_telescope_list(crs_name="corsika")
+
+        # Building a second layout from the file exported by the first one
+        layout_2 = LayoutArray(site=site_label, mongo_db_config=db_config, name="test_layout_2")
```

Comment:
I think we saw that you don't need to provide the site and mongoDB info if you provide instead the file in the init, no?

Reply:
yes, you are right. This was done before your PR and I forgot to adjust to it. Now I removed the two parameters and the test still passes.